src/aat/java/uk/gov/hmcts/ccd/datastore/tests/functional/elasticsearch/ElasticsearchCaseSearchTest.java

```diff
@@ -0,0 +1,157 @@
+
+            response.body("cases.size()", is(1));
+            response.body("cases[0].state", is(State.IN_PROGRESS));
+        }
+    }
+
+    @Nested
+    @DisplayName("Exact match")
+    class ExactMatch {
+
+        @Nested
+        @DisplayName("text field")
+        class TextField {
+
+            @Test
+            @DisplayName("should return case for exact match on a text field")
+            void shouldReturnCaseForExactMatchOnTextField() {
+                searchCaseForExactMatchAndVerifyResponse("TextField", TEXT);
+            }
+        }
+
+        @Nested
+        @DisplayName("date time field")
+        class DateTimeField {
+
+            @Test
+            @DisplayName("should return case for exact match on a date time field")
+            void shouldReturnCaseForExactMatchOnDateTimeField() {
+                searchCaseForExactMatchAndVerifyResponse("DateTimeField", DATE_TIME);
+            }
+        }
+
+        private void searchCaseForExactMatchAndVerifyResponse(String field, String value) {
+            String jsonSearchRequest = createExactMatchSearchRequest(field, value);
+
+            ValidatableResponse response = searchCaseAsPrivateCaseWorker(jsonSearchRequest);
+
+            verifyExactMatchResponse(response, field, value);
+        }
+
+        private String createExactMatchSearchRequest(String field, String value) {
+            return "{"
+                + "  \"query\": {"
+                + "    \"match\": {"
+                + "    \"data." + field + "\" : \"" + value + "\""
+                + "    }"
+                + "  }"
+                + "}";
+        }
+
+        private void verifyExactMatchResponse(ValidatableResponse response, String field, String expectedValue) {
+            response.body("cases.size()", is(1));
+            response.body("cases[0].case_data." + field, is(expectedValue));
+        }
+    }
+
+    @Nested
+    @DisplayName("Wildcard")
+    class Wildcard {
+
+        @Nested
+        @DisplayName("text field")
+        class TextField {
+
+            @Test
+            @DisplayName("should return case matching wildcard expression on a text field")
+            void shouldReturnCaseForWildcardMatchOnTextField() {
+                String wildcardExpr = TEXT.substring(0, 3).toLowerCase() + "*";
+                searchCaseByWildcardAndVerifyResponse("TextField", wildcardExpr, TEXT);
+            }
+        }
+
+        private void searchCaseByWildcardAndVerifyResponse(String field, String wildcardExpr, String expectedValue) {
+            String jsonSearchRequest = createWildcardSearchRequest(field, wildcardExpr);
+
+            ValidatableResponse response = searchCaseAsPrivateCaseWorker(jsonSearchRequest);
+
+            verifyWildcardMatchResponse(response, field, expectedValue);
+        }
+
+        private String createWildcardSearchRequest(String field, String wildcardExpr) {
+            return "{"
+                + "  \"query\": {"
```

Comment:
can we move this to ElasticSearchQueries?

Reply:
done